### PR TITLE
Hooli-style Popup

### PR DESCRIPTION
**Pull request by denismaysov**
### Hooli-style Popup

[Demo](https://denismaysov.github.io/Hooli-PopUp/) |
[Code base](https://github.com/denismaysov/Hooli-PopUp)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/denismaysov/html-css-popup/style.css [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+.header__popup__label_more {
+    display: flex;
+	 background-color: #424242;
+	 justify-content: center;
+    padding: 9px 0px;
+    color: #fff;
+    font-size: 18px;
+	 border: 1px solid #3F3F3F;
+    border-radius: 3%;
+}
```
Comment (A-Ostrovnyy) [inline submissions/denismaysov/html-css-popup/style.css:153]: Use [prettier](https://prettier.io/) to fix code indentation

File: submissions/denismaysov/html-css-popup/index.html [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+            <nav class="header__menu">
+                <ul class="header__list_items">
+                    <li class="header__list_item"><a href="#" class="header__link link focus_item">Mail</a></li>
+                    <li class="header__list_item"><a href="#" class="header__link link focus_item">Images</a></li>
+                    <li class="header__list_item">
+                        <input type="checkbox" id="popup_checkbox" class="header__popup_checkbox">
+                        <label for="popup_checkbox" class="header__popup_label">
+                            <img src="assets/popup-button.png" width="25" height="25" alt="">
+                        </label>
+                        <div class="header__popup invisible-scrollbar">
```
Comment (A-Ostrovnyy) [inline submissions/denismaysov/html-css-popup/index.html:21]: Add mobile version.
image

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good job!

**Comment by denismaysov** [comment]
can you help me with pushing my changes? I moved the project folder to another location and when I want to push or add changes, it gives me an error that you are not in the git rep, how to fix it

File: submissions/denismaysov/html-css-popup/index.html [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+                            <ul class="header__popup_items">
+                                <li class="header__popup_item">
+                                    <a href="#" class="header__popup_link link">
+                                        <img src="assets/mail.png" width="48" height="35" alt="#"
```
Comment (A-Ostrovnyy) [inline submissions/denismaysov/html-css-popup/index.html:25]: Read how and when to populate alt attribute value [Как правильно написать alt-текст](https://htmlacademy.ru/blog/articles/alt-text)
image

File: submissions/denismaysov/html-css-popup/style.css (hunk 1 above)
Comment (A-Ostrovnyy) [reply submissions/denismaysov/html-css-popup/style.css]: Still not fixed :cry:

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good progress!

**Comment by denismaysov** [comment]
Thx, when this pr will be done?

**Comment by A-Ostrovnyy** [comment]
@denismaysov when you push new changes according to review comments and I accept them

**Comment by denismaysov** [comment]
heh, i use pretier in vsc but here its not working. why?

**Comment by A-Ostrovnyy** [comment]
I suppose that you use different indentation entities, in some lines, you use `tab` in other `space`.If you use VS Code [try this](https://stackoverflow.com/questions/34174207/how-to-change-indentation-in-visual-studio-code) if not find something like this for your editor

**Comment by denismaysov** [comment]
I commit changes, but in my project created new branch. Why? There 2 branches now main and master. In repo hooli-popup

**Comment by denismaysov** [comment]
image

File: submissions/denismaysov/html-css-popup/.prettierrc.extension.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+
```
Comment (A-Ostrovnyy) [inline submissions/denismaysov/html-css-popup/.prettierrc.extension.json:1]: This file is not for review so remove him from the current PR

File: submissions/denismaysov/html-css-popup/index.html [hunk 5]
```diff
@@ -0,0 +1,299 @@
+<!DOCTYPE html>
+<html lang="ru">
```
Comment (A-Ostrovnyy) [inline submissions/denismaysov/html-css-popup/index.html:2]: You set russian language as the main page language but you use russian only as `alt` attribute value. Also, icons are decorative images, and decorative images must have empty `alt` value.

File: submissions/denismaysov/html-css-popup/index.html [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+                type="checkbox"
+                id="popup_checkbox"
+                class="header__popup_checkbox"
+              />
+              <label for="popup_checkbox" class="header__popup_label">
+                <img
+                  src="assets/popup-button.png"
+                  width="25"
+                  height="25"
+                  alt="Кнопка которая открывает Попап"
```
Comment (A-Ostrovnyy) [inline submissions/denismaysov/html-css-popup/index.html:31]: If you want to inform the user you should move this text to `aria-label` attribute of the parent label.

File: submissions/denismaysov/html-css-popup/index.html [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+                        />
+                        Chat
+                      </a>
+                    </li>
+                    <li class="header__popup_item">
+                      <a href="#" class="header__popup_link link">
+                        <img
+                          src="assets/blogs.png"
+                          width="48"
+                          height="35"
```
Comment (A-Ostrovnyy) [inline submissions/denismaysov/html-css-popup/index.html:222]: :fire:

**Review by A-Ostrovnyy — COMMENTED** [review]
Good job!
If you have some problems with git or with something else don't hesitate and text to chat in telegram or discord.

**Comment by denismaysov** [comment]
Soooo.... I fixed alts again and delete jsob file. What else?

File: submissions/denismaysov/html-css-popup/index.html [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+              </label>
+              <div class="header__popup invisible-scrollbar">
+                <ul class="header__popup_items">
+                  <li class="header__popup_item">
+                    <a href="#" class="header__popup_link link">
+                      <img
+                        src="assets/mail.png"
+                        width="48"
+                        height="35"
+                        alt="Mail"
```
Comment (A-Ostrovnyy) [inline submissions/denismaysov/html-css-popup/index.html:42]: Read how and when to populate alt attribute value [Как правильно написать alt-текст](https://web.archive.org/web/20220812092150/https://htmlacademy.ru/blog/articles/alt-text)

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Last issue with `alt` attribute value

**Comment by denismaysov** [comment]
I read this article again and again correct mistakes :D

**Comment by denismaysov** [comment]
What this button do?
image

**Comment by denismaysov** [comment]
RE-reques review

File: submissions/denismaysov/html-css-popup/index.html [hunk 9]
```diff
@@ -0,0 +1,294 @@
+<!DOCTYPE html>
+<html lang="en">
```
Comment (A-Ostrovnyy) [inline submissions/denismaysov/html-css-popup/index.html:2]: You set the main language of your page as "English" but you use russian. Also, try to check your text on typos
image

File: submissions/denismaysov/html-css-popup/index.html [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+              </label>
+              <div class="header__popup invisible-scrollbar">
+                <ul class="header__popup_items">
+                  <li class="header__popup_item">
+                    <a href="#" class="header__popup_link link">
+                      <img
+                        src="assets/mail.png"
+                        width="48"
+                        height="35"
+                        alt="Електронная почта"
```
Comment (A-Ostrovnyy) [inline submissions/denismaysov/html-css-popup/index.html:42]: Icons are decorative elements
image

File: submissions/denismaysov/html-css-popup/index.html [hunk 11]
```diff
@@ -0,0 +1,294 @@
...
+              >
+                <img
+                  src="assets/favicon.png"
+                  class="header__photo focus_item"
+                  width="25"
+                  height="25"
+                  alt=""
+                />
+              </a>
+            </li>
```
Comment (A-Ostrovnyy) [inline submissions/denismaysov/html-css-popup/index.html:288]: Those images are equal by role and functionality, why did you remove the `alt`  attribute value only in one image?

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]

**Comment by A-Ostrovnyy** [comment]
> What this button do? image

With this button, you notify the reviewer that your code is ready for review.

**Comment by denismaysov** [comment]
I must use alts on en, this icons does not carry any meaning. I must use alts on that icons?

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
1. Use the same language for the `alt` attribute and for the link content.
2. For images inside links in the popup you set an `alt` value, but for the images inside links on the top level menu(lines 269 - 289) you set an empty value. Try to use a single approach for similar entities). Icons are decorative images no need to populate `alt` attribute for decorative elements.
By default, the focus goes to interactive elements so if you want to indicate to the user the destination or other information use the `aria-label` attribute for interactive elements.
```HTML
<a aria-label="mail"></a>
<button aria-label="mail"></button>
```

**Comment by denismaysov** [comment]
Done 
I have a question, how can I do changin color in svg icon by the click without js

**Comment by denismaysov** [comment]
what else should i do?

File: submissions/denismaysov/html-css-popup/index.html [hunk 12]
```diff
@@ -0,0 +1,295 @@
...
+                type="checkbox"
+                id="popup_checkbox"
+                class="header__popup_checkbox"
+              />
+              <label for="popup_checkbox" class="header__popup_label">
+                <img
+                  src="assets/popup-button.png"
+                  width="25"
+                  height="25"
+                  alt="Open Popup"
```
Comment (A-Ostrovnyy) [inline submissions/denismaysov/html-css-popup/index.html:31]: The icon is a decorative image, set an empty value to `alt` attribute
```HTML
<img alt="" >
```

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
One last issue.
Just set an empty value to `alt` attribute of icons.

**Comment by denismaysov** [comment]
Done!

**Comment by denismaysov** [comment]
Can u check my commits in my multiPopup? It`s my penalty task.

**Comment by denismaysov** [comment]
When u merge my popup?)

File: submissions/denismaysov/html-css-popup/index.html [hunk 13]
```diff
@@ -0,0 +1,295 @@
...
+                class="header__link focus_item"
+              >
+                <img
+                  src="assets/favicon.png"
+                  class="header__photo focus_item"
+                  width="25"
+                  height="25"
+                  aria-label="MyGitHub"
+                />
+              </a>
```
Comment (A-Ostrovnyy) [inline submissions/denismaysov/html-css-popup/index.html:288]: The `aria-label` attribute better move to the link instead of image. Also, add an `alt` attribute with an empty value, because `alt` is a required attribute.

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
One last issue.

**Review by A-Ostrovnyy — APPROVED** [review]
Well done!
